fgpyo/sam/__init__.py

```diff
@@ -876,6 +876,78 @@ def set_mate_info(
     r2.is_proper_pair = proper_pair
 
 
+def set_mate_info_on_secondary(
+    primary: AlignedSegment,
+    secondary: AlignedSegment,
```

Comment:
I think I would call the first argument "mate_primary" or something to make it clear you're supposed to provide the primary of the mate, not the primary of the secondary.